src/clr-angular/utils/i18n/common-strings.service.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import { Injectable, SkipSelf, Optional } from '@angular/core';
+
+import { ClrCommonStrings } from './common-strings.interface';
+
+@Injectable()
+export class ClrCommonStringsService implements Required<ClrCommonStrings> {
+  open = 'Open';
+  close = 'Close';
+  show = 'Show';
+  hide = 'Hide';
+  expand = 'Expand';
+  collapse = 'Collapse';
+  more = 'More';
+  select = 'Select';
+  selectAll = 'Select All';
+  previous = 'Previous';
```

Comment:
Wouldn't we want more descriptive terms, like 'Previous page' or 'Close alert'. If so, we wouldn't want to reuse these strings and they would be specific to each icon use. When you translate a string I'm not certain that in each language or situation that it could be generalized. 

I'm thinking we'd want a property for each instance, and be more descriptive by default.

Reply:
I considered it a long time as I was implementing this, I agree I'd much rather have more specific labels. It's clearly better UX for people with disabilities.

The problem is that it bloats our i18n API to a point where you have dozens and potentially hundreds of strings in the future, which becomes horrible to provide for our consumers. I think even right now we have around 50 of them, on a first crude pass.

It's still an interesting topic and I would like to find a better solution, whether we implement it right now or postpone it because of the workload. Let's talk about this offline.